fix(transformer/class-properties): fix `ScopeId`s in static prop initializers

overlookmotel opened the pull request:
Code in static property initializers moves from inside the class to outside. Update parent `ScopeId`s for first level scopes in initializers.

overlookmotel commented:
* **#7791** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7791?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7791?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
That's huge!

Dunqing commented:
### Merge activity

* **Dec 10, 10:17 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(transformer/class-properties): fix `ScopeId`s in static prop initializers (#7791)

Code in static property initializers moves from inside the class to outside. Update parent `ScopeId`s for first level scopes in initializers.
